extra/src/main/java/aima/extra/probability/BigDecimalProbabilityNumber.java

```diff
@@ -0,0 +1,381 @@
+package aima.extra.probability;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.math.MathContext;
+import java.math.RoundingMode;
+
+/**
+ * Implementation based on the underlying BigDecimal datatype. Supports higher
+ * precision than DoubleProbabilityNumber (DECIMAL128 precision by default, but
+ * can go upto UNLIMITED). The class is immutable.
+ * 
+ * @author Nagaraj Poti
+ *
+ */
+public class BigDecimalProbabilityNumber implements ProbabilityNumber, Comparable<ProbabilityNumber> {
+
+	// Constants
+
+	/**
+	 * IEEE 754R Decimal128 format - Precision of 34 digits and a rounding mode
+	 * of HALF_EVEN. HALF_EVEN rounding mode statistically minimizes cumulative
+	 * error when applied repeatedly over a sequence of calculations.
+	 */
+	private static final Integer DEFAULT_PRECISION = MathContext.DECIMAL128.getPrecision();
+
+	private static final RoundingMode DEFAULT_PRECISION_ROUNDING_MODE = MathContext.DECIMAL128.getRoundingMode();
+
+	/**
+	 * Maximum integer value that can be accomodated as exponent in the
+	 * BigDecimal pow() function
+	 */
+	private static final int EXPONENT_MAX = 999999999;
+
+	// Internal fields
+
+	private BigDecimal value;
+
+	/**
+	 * MathContext of value.
+	 */
+	private MathContext precisionMathContext = new MathContext(DEFAULT_PRECISION, DEFAULT_PRECISION_ROUNDING_MODE);
+
+	private BigDecimal BIG_DECIMAL_ZERO = new BigDecimal(0, precisionMathContext);
+
+	private BigDecimal BIG_DECIMAL_ONE = new BigDecimal(1, precisionMathContext);
+
```

Comment:
It likely makes more sense to declare these two constants as static and to give them unlimited precision (this will save a lot of space and address inconsistencies across different instances with different precisions).